src/lyra.ts

```diff
@@ -292,7 +292,8 @@ function recursiveradixInsertion<S extends PropertiesSchema>(
           tokenOccurrencies[propName][token] = 0;
         }
 
-        tokenOccurrencies[propName][token]++;
+        // increase a token counter that may not yet exist
+        tokenOccurrencies[propName][token] = (tokenOccurrencies[propName][token] ?? 0) + 1;
```

Comment:
Consider the following, taken from [lyra.ts](https://github.com/LyraSearch/lyra/blob/0a910977c174c59d993618d4c67a2d61617966cd/src/lyra.ts#L295):

```typescript
const propName = 'prop';
const token = 'doc-id';
const tokenOccurrencies = { [propName]: { } };
```

What happens if we increase a non-existing counter like follows?

```typescript
tokenOccurrencies[propName][token]++;
```

Two things happen:
- `tokenOccurrencies[propName]` becomes `{ 'doc-id': NaN }`
- `NaN` is returned from the postfix increment expression

If we wanted to increment a counter that may not yet exist, we should instead run something like:

```typescript
tokenOccurrencies[propName][token] = (tokenOccurrencies[propName][token] ?? 0) + 1;
```

which:
- sets the counter to 1 if it didn't exist
- increment the counter by 1 otherwise

Reply:
Damn you're right. Thank you so much!